webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala

```diff
@@ -206,6 +206,26 @@ class FindExpectedReportsJdbcRepository(
     }
   }
 
+  /**
+   * Return node ids associated to the directive (based on expectedreports (the one still pending)) for this Directive,
+   * only limited on the nodeIds in parameter (used when cache is incomplete)
+   */
+  override def findCurrentNodeIdsForDirective(directiveId: DirectiveId, nodeIds: Set[NodeId]): IOResult[Set[NodeId]] = {
```

Comment:
why do you query the database when you have a cache of current expected reports?
in the NodeConfigurationService, you could do the same as findNodesApplyingRule but with a directive
it would be much more efficient as data is already there